test/e2e/apimachinery/health_handlers.go

```diff
@@ -0,0 +1,141 @@
+		"[+]poststarthook/apiservice-status-available-controller ok",
+		"[+]poststarthook/kube-apiserver-autoregistration ok",
+		"[+]autoregister-completion ok",
+		"[+]poststarthook/apiservice-openapi-controller ok",
+	}
+	requiredLivezChecks = []string{
+		"[+]ping ok",
+		"[+]log ok",
+		"[+]etcd ok",
+		"[+]poststarthook/start-kube-apiserver-admission-initializer ok",
+		"[+]poststarthook/generic-apiserver-start-informers ok",
+		"[+]poststarthook/start-apiextensions-informers ok",
+		"[+]poststarthook/start-apiextensions-controllers ok",
+		"[+]poststarthook/crd-informer-synced ok",
+		"[+]poststarthook/bootstrap-controller ok",
+		"[+]poststarthook/rbac/bootstrap-roles ok",
+		"[+]poststarthook/scheduling/bootstrap-system-priority-classes ok",
+		"[+]poststarthook/start-cluster-authentication-info-controller ok",
+		"[+]poststarthook/aggregator-reload-proxy-client-cert ok",
+		"[+]poststarthook/start-kube-aggregator-informers ok",
+		"[+]poststarthook/apiservice-registration-controller ok",
+		"[+]poststarthook/apiservice-status-available-controller ok",
+		"[+]poststarthook/kube-apiserver-autoregistration ok",
+		"[+]autoregister-completion ok",
+		"[+]poststarthook/apiservice-openapi-controller ok",
+	}
+	requiredReadyzChecks = []string{
+		"[+]ping ok",
+		"[+]log ok",
+		"[+]etcd ok",
+		"[+]poststarthook/start-kube-apiserver-admission-initializer ok",
+		"[+]poststarthook/generic-apiserver-start-informers ok",
+		"[+]poststarthook/start-apiextensions-informers ok",
+		"[+]poststarthook/start-apiextensions-controllers ok",
+		"[+]poststarthook/crd-informer-synced ok",
+		"[+]poststarthook/bootstrap-controller ok",
+		"[+]poststarthook/rbac/bootstrap-roles ok",
+		"[+]poststarthook/scheduling/bootstrap-system-priority-classes ok",
+		"[+]poststarthook/start-cluster-authentication-info-controller ok",
+		"[+]poststarthook/aggregator-reload-proxy-client-cert ok",
+		"[+]poststarthook/start-kube-aggregator-informers ok",
+		"[+]poststarthook/apiservice-registration-controller ok",
+		"[+]poststarthook/apiservice-status-available-controller ok",
+		"[+]poststarthook/kube-apiserver-autoregistration ok",
+		"[+]autoregister-completion ok",
+		"[+]poststarthook/apiservice-openapi-controller ok",
+		"[+]informer-sync ok",
+	}
+)
+
+func testPath(client clientset.Interface, path string, requiredChecks []string) error {
+	var result restclient.Result
+	err := wait.Poll(100*time.Millisecond, 30*time.Second, func() (bool, error) {
+		result = client.CoreV1().RESTClient().Get().RequestURI(path).Do(context.TODO())
+		status := 0
+		result.StatusCode(&status)
+		return status == 200, nil
+	})
+	if err != nil {
+		return err
+	}
+	body, err := result.Raw()
+	if err != nil {
+		return err
+	}
+	checks := sets.NewString(strings.Split(string(body), "\n")...)
+	for _, check := range requiredChecks {
+		if !checks.Has(check) {
+			return fmt.Errorf("required %s check: %s not present: %s", path, check, string(body))
+		}
+	}
```

Comment:
nit: testing if `requiredChecks.Difference(checks)` is non-empty and logging all missing required checks avoids iterating on CI adding a single string each time